src/QuickMarcEditor/QuickMarcDuplicateWrapper.js

```diff
@@ -59,31 +59,31 @@ const QuickMarcDuplicateWrapper = ({
   };
 
   const getQuickMarcRecordStatus = (qmRecordId) => {
-    let isFirstRequest = true;
+    let requestCount = 1;
+    let intervalId;
 
     function makeRequest() {
       mutator.quickMarcRecordStatus.GET({ params: { qmRecordId } })
         .then(({ instanceId, status }) => {
           if (status === 'ERROR') {
+            clearInterval(intervalId);
             showCallout({
               messageId: 'ui-quick-marc.record.saveNew.error',
               type: 'error',
             });
           }
 
           if (status === 'IN_PROGRESS') {
-            if (!isFirstRequest) {
+            if (requestCount === 4) {
```

Comment:
Let's add another constant like `QM_RECORD_STATUS_BAIL_TIME` equal to `20000`
and a variable like `maxRequestAttempts` that would be equal to `QM_RECORD_STATUS_BAIL_TIME / QM_RECORD_STATUS_TIMEOUT`

I think this will be more understandable than just `4`